texar/torch/run/executor.py

```diff
@@ -1369,8 +1374,12 @@ def test(self, dataset: OptionalDict[DatasetBase] = None):
                 If `None`, :attr:`test_data` from the constructor arguments is
                 used. Defaults to `None`.
         """
-        # open the log files
-        self._open_files()
+        # Check whether files have been opened, to avoid re-opening and closing.
+        # This could happen when, e.g., `test` is called in a registered hook
+        # during training.
+        should_open_file = (len(self._opened_files) == 0)
```

Comment:
In this case, we won't open a file when we call `test` inside `train`. Where will we store the outputs of the testing procedure? Do we append the testing results to the file that is opened in the training procedure?

Reply:
If we call `test` within `train` then the files should already be open at the beginning of the `train` method, and won't be closed until the end of `train`. So logging would work as usual.